server/parameters/util/YAMLParser.java

```diff
@@ -398,5 +403,38 @@ private static boolean isValidSizeString(String size) {
                 return matcher.matches();
             }
         }
+
+        private static class Duration {
+
+            private final static String LENGTH_PART = "(^[0-9]+)";
+            private final static int LENGTH_GROUP = 1;
+            private final static String UNIT_PART = "([wdhms]$)";
+            private final static int UNIT_GROUP = 2;
+            private static final Pattern DURATION_PATTERN = Pattern.compile(LENGTH_PART + UNIT_PART, Pattern.CASE_INSENSITIVE);
+
+            private static long parse(String duration) {
+                Matcher matcher = DURATION_PATTERN.matcher(duration);
+                long coefficient;
+                if (matcher.matches()) {
+                    String lenStr = matcher.group(LENGTH_GROUP);
+                    String unitStr = matcher.group(UNIT_GROUP);
+                    long lenValue = Long.parseLong(lenStr);
+                    if (unitStr.equalsIgnoreCase("w")) coefficient = com.vaticle.typedb.core.common.collection.Duration.WEEK;
```

Comment:
can we avoid creating our own constants and either use java's `Duration` directly, or it's `ChronoUnit`?

Reply:
We can, but we'd have to be happy to not support weeks. Java doesn't feel that weeks is an exact duration:
```
external/vaticle_typedb/server/parameters/util/YAMLParser.java:424: error: [DurationTemporalUnit] Duration APIs only work for DAYS or exact durations.
                    if (unitStr.equalsIgnoreCase("w")) result = java.time.Duration.of(lenValue, ChronoUnit.WEEKS).toSeconds();
                                                                                     ^
```
https://errorprone.info/bugpattern/DurationTemporalUnit